File: assignment2/VerifyGaussSeidel.m
function err = VerifyGaussSeidel(n)

% Generate a random matrix
A = rand(n);
b = rand(n,1);

% Make the matrix diagonally dominant
for i=1:n
  c = 0;
  for j=1:n
    c = c + A(i,j);
  end
  A(i,i) = c;
end

% Solve using Gauss-Seidel
x = GaussSeidel(A, zeros(n,1), b, 1e-12, 1000);

% Compare with the built-in MATLAB solver
err = norm(x - A\b);

File: assignment2/VerifyGaussSeidel.py
import numpy as np
from GaussSeidel import GaussSeidel

n = 100

A = np.random.rand(n, n)
b = np.random.rand(n)

for i in range(n):
    A[i, i] = sum(A[i, j] for j in range(n))

x = GaussSeidel(A, np.zeros(n), b, 1e-12, 1000)

error = np.linalg.norm(x - np.linalg.solve(A, b))

print(f"total error: {error}")
